firmware/library/L2_HAL/temperature.hpp

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <cstdint>
+#include <cstdio>
+#include "L1_Drivers/i2c.hpp"
+
+class TemperatureInterface
+{
+ public:
+  virtual bool Initialize()                       = 0;
+  virtual int GetCelsius()                        = 0;
+  virtual int GetFahrenheit(int temperature_data) = 0;
```

Comment:
```suggestion
  virtual int GetFahrenheit() = 0;
```

Reply:
Did all the changes and moved the file over to L2_HAL/sensors/temperature.  Also removed the example main.cpp but couldn't remove the folder. Should be fine there I'll include the example after implementation is pull.